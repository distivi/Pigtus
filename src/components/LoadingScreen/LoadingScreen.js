import React from 'react';
import PropTypes from 'prop-types';
import { ProgressBar } from 'react-bootstrap';

import { connect } from 'react-redux';

export class LoadingScreen extends React.Component {
  constructor() {
    super();

    this.appName = 'PIGTUS';
  }

  render() {
    return (
      <div className="loading-screen">
        <div className="loading-container">
          <div className="app-name"> {this.appName} </div>
          <div className="loading-progress">
            <ProgressBar min={0} max={100} now={this.props.loadingPercent} />
          </div>
        </div>
      </div>
    );
  }
}

LoadingScreen.propTypes = {
  loadingPercent: PropTypes.number
};

LoadingScreen.defaultProps = {
  loadingPercent: 0
};

const mapStateToProps = (state) => {
  return {
    loadingPercent: state.loading.loadingPercent
  };
};

export default connect(mapStateToProps)(LoadingScreen);
