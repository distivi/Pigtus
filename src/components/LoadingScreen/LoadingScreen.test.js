import React from 'react';
import ReactDOM from 'react-dom';
import { LoadingScreen } from './LoadingScreen';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { ProgressBar } from 'react-bootstrap';

describe('<LoadingScreen/>', () => {
  it('renders LoadingScreen without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LoadingScreen />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('LoadingScreen is consistent with mockup', () => {
    const component = renderer.create(<LoadingScreen loadingPercent={10} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('LoadingScreen app name exists', () => {
    const expected = 'PIGTUS';

    const appNameClass = '.app-name';

    const loadingScreen = shallow(<LoadingScreen />);
    const appName = loadingScreen.find(appNameClass).text();

    expect(appName.length).toBeTruthy();
    expect(appName.replace(/ /g, '')).toBe(expected);
  });

  it('LoadingScreen progress-bar width equal to 20 percent', () => {
    const expected = '20%';
    const progressBarClass = '.progress-bar';

    const loadingScreen = shallow(<LoadingScreen loadingPercent={20} />);

    const progressBar = loadingScreen.find(ProgressBar).shallow();

    expect(progressBar.length).toBe(1);
    const progressBarWidth = progressBar.find(progressBarClass).get(0).props
      .style.width;

    expect(progressBarWidth).toBe(expected);
  });
});
