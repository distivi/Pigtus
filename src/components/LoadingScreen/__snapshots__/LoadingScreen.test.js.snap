// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingScreen/> LoadingScreen is consistent with mockup 1`] = `
<div
  className="loading-screen"
>
  <div
    className="loading-container"
  >
    <div
      className="app-name"
    >
       
      PIGTUS
       
    </div>
    <div
      className="loading-progress"
    >
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={10}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "10%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
